### DISCUSSION TITLE
Rotate around World Axis instead of Relative One

### DISCUSSION THREAD

**Discussion by N8python** [enable3d/enable3d#170] [Help / Questions]
So, I have been making a simple flight controller. This involves mutating the plane's X, Y, and Z rotations:
```js
this.player.rotation.y += this.player.velocity.rotation.y;
this.player.rotation.z += this.player.velocity.rotation.z;
this.player.rotation.x += this.player.velocity.rotation.x;
```
However, when I rotate the player on the y axis, its x axis rotates as well and it doesn't rotate directly on the words x axis - it instead rotates on its local x axis.

How can I make the X/Y rotation of the player invariant to already existing rotation?

**Comment by yandeu** [comment]
I believe there is a topic about flight controller in the three.js forum?

**Reply by N8python** [reply]
I didn't know - do you have a link?
